cmake/FindDPCPP.cmake

```diff
@@ -23,7 +23,7 @@ set(DPCPP_FLAGS "-fsycl;-sycl-std=2020;${DPCPP_FLAGS}")
 # -fsycl-id-queries-fit-in-int is an optimization enabled by default, but
 # adds non-conformant behavior that limits the number of work-items in an
 # invocation of a kernel, so we disable this behavior here.
-set(DPCPP_FLAGS "${DPCPP_FLAGS};-fno-sycl-id-queries-fit-in-int")
+# set(DPCPP_FLAGS "${DPCPP_FLAGS};-fno-sycl-id-queries-fit-in-int")
```

Comment:
As noted in the comments above this option is required to disable optimization, which technically does not follow the SYCL standard.

I see two changes in `cmake/FindDPCPP.cmake`. Are both these changes required to overcome the problem you mentioned in the [PR comment](https://github.com/KhronosGroup/SYCL-CTS/pull/872#issuecomment-2562662012)?

> moving function prototypes around messed with overload resolution in DPC++ (not sure if the code was actually incorrect or if I was triggering a compiler bug).

I would prefer to investigate this issue to understand the root cause of it. Most of the DPC++ team is holidays, but @AlexeySachkov, @steffenlarsen and @KornevNikita should be able to help when they return.

Conceptually these changes should not be required to disable tests using `sycl::half` data type.

Reply:
Whoops, this was a leftover from me debugging my local DPC++ setup. Reverted.